src/optimisation/Optimisation.jl

```diff
@@ -83,6 +83,12 @@ function DynamicPPL.dot_tilde_assume(ctx::OptimizationContext, right, left, vns,
     return r, lp, vi
 end
 
+DynamicPPL.tilde_observe(::OptimizationContext{<:PriorContext}, right, left, vi) =
+    (0, vi)
+
+DynamicPPL.dot_tilde_observe(::OptimizationContext{<:PriorContext}, right, left, vi) =
+    (0, vi)
```

Comment:
```suggestion
DynamicPPL.tilde_observe(ctx::OptimizationContext{<:PriorContext}, args...) =
    DynamicPPL.tilde_observe(ctx.context, args...)

DynamicPPL.dot_tilde_observe(ctx::OptimizationContext{<:PriorContext}, args...) =
    DynamicPPL.dot_tilde_observe(ctx.context, args...)
```
I think falling back on the `PriorContext` implementation is a bit neater. Result is the same.